assets/src/components/mapPage.tsx

```diff
@@ -76,6 +74,67 @@ const SearchInputAndResults = ({
   )
 }
 
+const SelectedVehicle = ({ vehicleId }: { vehicleId: VehicleId }) => {
+  // TODO: When using socket from context, this doesn't work as-is
+  // Presumably because the useMostRecentVehicleById hook is being used twice, but
+  // haven't tracked down error yet
+  const { socket } = useSocket()
+
+  const selectedVehicleOrGhost = useMostRecentVehicleById(socket, vehicleId)
+
+  return selectedVehicleOrGhost ? (
+    <VehiclePropertiesCard
+      vehicleOrGhost={selectedVehicleOrGhost}
+      key={selectedVehicleOrGhost.id}
+    />
+  ) : (
+    <Loading />
+  )
+}
+
+const Selection = ({
+  selectedEntity,
+  setSelection,
+}: {
+  selectedEntity: SelectedEntity
+  setSelection: (selectedEntity: SelectedEntity | null) => void
+}): ReactElement => {
+  const [{ searchPageState }, dispatch] = useContext(StateDispatchContext)
+  return (
+    <div>
+      <div className="c-map-page__search-actions">
+        {searchPageState.query.text !== "" && (
+          <button
+            className="c-map-page__back-button"
+            onClick={() => {
+              setSelection(null)
+            }}
+          >
+            <ChevronLeftIcon />
+            Back
+          </button>
+        )}
+        <button
+          className="button-submit c-map-page__new-search-button"
+          onClick={() => {
+            setSelection(null)
+            dispatch(clearSearch())
+          }}
+        >
+          <SearchIcon />
+          New Search
+        </button>
+      </div>
+      <hr />
+      {selectedEntity.type === SelectedEntityType.Vehicle ? (
+        <SelectedVehicle vehicleId={selectedEntity.vehicleId} />
+      ) : (
+        <p>TODO: Route selection</p>
```

Comment:
That is correct! From discussion with product and design during [review](https://app.asana.com/0/1200180014510248/1204729792351246/f), the plan is to not merge this PR until the RPC changes are also in place. Would it be preferable to create a new branch to merge this into rather than master that we can merge all the related changes into?

Reply:
Yeah, I guess my concern is that, as discussed on Slack yesterday, there's actually a fair bit of interdependence between this work and the RPC work, so we probably need to either accept that one will go live slightly before the other or create a shared feature branch to merge these changes into.

Reply:
👍 I'd be down to try the shared feature branch approach so that it is clearer what work is considered "done" and ready to start building on. Will add one now!